File: src/service/AccountService.java
package service;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

import domain.Account;

public class AccountService {
	
	private List<Account> accounts;
	private static AccountService self = new AccountService();
	
	public static AccountService getInstance() {
		return self;
	}
	
	private AccountService() {
		accounts = new ArrayList<Account>();
		
		Account account = new Account();
		account.setAccountId(1001);
		account.setFirstName("Joe");
		account.setLastName("Smith");
		account.setCurrentBalance(new BigDecimal("32.00"));
		accounts.add(account);
		
		account = new Account();
		account.setAccountId(1002);
		account.setFirstName("John");
		account.setLastName("Doe");
		account.setCurrentBalance(new BigDecimal("22.00"));
		accounts.add(account);

		account = new Account();
		account.setAccountId(1003);
		account.setFirstName("Jane");
		account.setLastName("Doe");
		account.setCurrentBalance(new BigDecimal("5.00"));
		accounts.add(account);
	}

	public List<Account> getAccounts() {
		return accounts;
	}
	
	public List<Account> getAccount(long accountId) {
		return accounts.stream().filter((acct) -> acct.getAccountId() == accountId).collect(Collectors.toList());
	}
}

File: angularSrc/src/app/services/AccountService.ts
import { Injectable } from '@angular/core';
import {HttpModule, Http} from '@angular/http';

export class Account {
    constructor(
        public accountId: number,
        public firstName: string,
        public lastName: string,
        public currentBalance: number
    ) {}
}

@Injectable()
export class AccountService {
    accounts:  Array<Account>;
    currentLoggedInAccount: Account;

    constructor() {
        console.log('IN ACCOUNT SERVICE CONSTRUCTOR');
        this.accounts = initialAccounts.map(a => new Account(a.accountId,
            a.firstName, a.lastName, a.currentBalance));
        this.currentLoggedInAccount = this.accounts.pop();
    }

    getCurrentLoggedInAccount(): Account {
        return this.currentLoggedInAccount;
    }

    getCurrentBalance(): number {
        return this.currentLoggedInAccount.currentBalance;
    }
}

const initialAccounts = [
    {
        'accountId' : 1001,
        'firstName' : 'Joe',
        'lastName' : 'Smith',
        'currentBalance' : 32.00
    },
    {
        'accountId' : 1002,
        'firstName' : 'John',
        'lastName' : 'Doe',
        'currentBalance' : 22.00
    },
    {
        'accountId' : 1003,
        'firstName' : 'Jane',
        'lastName' : 'Doe',
        'currentBalance' : 5.00
    },
];
